### ISSUE TITLE
 'unexpected number of arguments' error at facl.rb:111:in `diff'

### ISSUE THREAD

**Issue by bby-bishopclark** [ncerny/facl#2]
the following piece of code
```  
    facl '/var/www/drupal7' do
      user        :'' => 'rwx'
      group       :'' => 'r-x',
         drupal:     'rwx'
      mask        :'' => 'rwx'
      other       :'' => 'r-x'
      only_if { ::Dir.exist?('/var/www/drupal7') }
    end
```

causes an error on chef 13/14 (14.1.1 used here):
``` 
ArgumentError: wrong number of arguments (given 0, expected 2)
/var/chef/cache/cookbooks/facl/resources/facl.rb:111:in `diff'
```

.. but it still makes the setfacl change.   A subsequent chef run, calculating no changes required, will succeed.

Is my invocation messed up?  Can YOU spot a problem?
(edit:  massage original code out of its each.do|d| bit)

**Comment by dtmmoreno** [comment]
I'm getting the same problem.  I was also doing the each.do for multiple directories.

**Comment by bby-bishopclark** [comment]
Bullwinkle:  Hey Rocky!  Watch me pull a bomb out of my hat!  

Rocky:  That trick never works.

Bullwinkle:
````
# because #badAddons
facl '/etc/shadow' do
  user          :'' => '---',
  jenkins: 'r--'
  group         :'' => '---'
  mask          :''=> '---'
  other         :'' => '---'
end
````

aaand 
````
Generated at 2018-09-14 14:42:27 -0700
ArgumentError: wrong number of arguments (given 0, expected 2)
/var/chef/cache/cookbooks/facl/resources/facl.rb:111:in `diff'
/opt/chef/embedded/lib/ruby/gems/2.5.0/gems/chef-14.4.56/lib/chef/data_collector/resource_report.rb:76:in `to_hash'
````

**Comment by Jamstah** [comment]
The chef code checks if the resource has a diff method and calls it in certain circumstances. The diff method needs renaming. Will open a PR, to fix that and some other bugs.

**Comment by Jamstah** [comment]
Fix is commit 77001d7029cd1172a8df36e93812757cc3b2aec0 in #3
